internal/util/diff/diff.go

```diff
@@ -102,34 +103,44 @@ func (c *Command) Run() error {
 		return errors.Errorf("package missing Kptfile at '%s': %v", c.Path, err)
 	}
 
-	// Stage current package
-	currPkg, err := ioutil.TempDir("", "kpt-")
+	// Create a staging directory to store all compared packages
+	stagingDirectory, err := ioutil.TempDir("", "kpt-")
 	if err != nil {
-		return errors.Errorf("failed to create stage dir for current package: %v", err)
+		return errors.Errorf("failed to create stage dir: %v", err)
 	}
 	defer func() {
+		// Cleanup staged content after diff. Ignore cleanup if debugging.
 		if !c.Debug {
-			defer os.RemoveAll(currPkg)
+			defer os.RemoveAll(stagingDirectory)
 		}
 	}()
 
+	// Stage current package
+	// This prevents prepareForDiff from modifying the local package
+	currPkg, err := stageDirectory(stagingDirectory, "local-package")
+	if err != nil {
+		return errors.Errorf("failed to create stage dir for current package: %v", err)
+	}
+
 	err = copyutil.CopyDir(c.Path, currPkg)
 	if err != nil {
 		return errors.Errorf("failed to stage current package: %v", err)
 	}
+	fmt.Printf("Staging %s at %s\n", c.Path, currPkg)
 
 	// get the upstreamPkg at current version
-	upstreamPkg, err := c.PkgGetter.GetPkg(kptFile.Upstream.Git.Repo,
+	upstreamPkg, err := c.PkgGetter.GetPkg(stagingDirectory,
+		kptFile.Upstream.Git.Repo,
 		kptFile.Upstream.Git.Directory,
 		kptFile.Upstream.Git.Commit)
+	fmt.Printf("Staging %s/%s@%s at %s\n",
+		kptFile.Upstream.Git.Repo,
+		kptFile.Upstream.Git.Directory,
+		kptFile.Upstream.Git.Commit,
+		upstreamPkg)
```

Comment:
same question as above.  Do people expect to see the directory where things are staged after the "at..."

Reply:
The `pkg` variables store the path to the directory containing the staged package. This may be refactored in the future. The path is returned from the `stageDirectory` and `GetPkg` functions.